PlaceRouteHierFlow/PnRDB/PnRdatabase.cpp

```diff
@@ -667,17 +667,20 @@ void PnRdatabase::CheckinChildnodetoBlock(PnRDB::hierNode& parent, int blockID,
   PnRDB::Omark ort = child.abs_orient;
   int width = child.UR.x - child.LL.x;
   int height = child.UR.y - child.LL.y;
-  PnRDB::point translate = parent.Blocks[blockID].instance[parent.Blocks[blockID].selectedInstance].placedBox.LL;
-  parent.Blocks[blockID].instance[parent.Blocks[blockID].selectedInstance].gdsFile = child.gdsFile;
+
+  auto& pi = parent.Blocks[blockID].instance[parent.Blocks[blockID].selectedInstance];
+
+  PnRDB::point translate = pi.placedBox.LL;
+  pi.gdsFile = child.gdsFile;
 
   // transform child blockpins orginals into placed in parent coordinate
   std::vector<PnRDB::pin> blockPins = child.blockPins;
   TransformBlockPinsOriginToPlaced(blockPins, translate, width, height, ort);
-  for (unsigned int p = 0; p < parent.Blocks[blockID].instance[parent.Blocks[blockID].selectedInstance].blockPins.size(); p++) {
```

Comment:
@854768750 This code copies blockPins into the parent. We shouldn't need to have a copy of the template for each instance (like we do in the top_down case). Or do you think otherwise?